File: src/prod/src/ServiceModel/management/FileStoreService/uploadsession.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace std;
using namespace Common;
using namespace Management::FileStoreService;

UploadSession::UploadSession()
: uploadSessions_()
{
}

UploadSession::UploadSession(
    std::vector<UploadSessionInfo> const & uploadSessions)
    : uploadSessions_(uploadSessions)
{
}

UploadSession& UploadSession::operator = (UploadSession const & other)
{
    if (this != &other)
    {
        this->uploadSessions_ = other.uploadSessions_;
    }

    return *this;
}

UploadSession& UploadSession::operator = (std::vector<UploadSessionInfo> const & session)
{
    this->uploadSessions_ = session;
    return *this;
}

UploadSession& UploadSession::operator = (UploadSession && other)
{
    if (this != &other)
    {
        this->uploadSessions_ = move(other.uploadSessions_);
    }

    return *this;
}

UploadSession& UploadSession::operator = (std::vector<UploadSessionInfo> && session)
{
    this->uploadSessions_ = std::move(session);
    return *this;
}

void UploadSession::WriteTo(TextWriter & w, FormatOptions const &) const
{
    w.Write("UploadSessionCount={0}", this->uploadSessions_.size());
}

File: src/prod/src/managed/Api/src/System/Fabric/query/UploadSession.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.Query
{
    using System.Fabric;
    using System.Fabric.Common.Serialization;
    using System.Fabric.Health;
    using System.Fabric.Interop;
    using System.Numerics;

    /// <summary>
    /// UploadSession contains information about the image store upload sessions
    /// </summary>
    public sealed class UploadSession
    {
        /// <summary>
        /// Initializes a new instance of the <cref name="UploadSession"/> structure
        /// </summary>
        /// <param name="uploadSessions">
        /// Get the upload session based on the given upload session ID or image store relative path
        /// </param>
        internal UploadSession(
            UploadSessionInfo[] uploadSessions)
        {
            this.UploadSessions = uploadSessions;
        }

        /// <summary>
        /// Initializes a new instance of the <cref name="UploadSession"/> structure
        /// </summary>
        internal UploadSession() { }

        /// <summary>
        /// <para>Get the upload session based on the given upload session ID or image store relative path.</para>
        /// </summary>
        /// <value>
        /// <para>The upload session.</para>
        /// </value>
        public UploadSessionInfo[] UploadSessions { get; private set; }
    }
}